apps/ledger-live-mobile/src/components/RootNavigator/ManagerNavigator.tsx

```diff
@@ -92,13 +81,39 @@ const Stack = createStackNavigator();
 export function ManagerTabIcon(props: any) {
   const isNavLocked = useIsNavLocked();
   const hasAvailableUpdate = useSelector(hasAvailableUpdateSelector);
+  const lastSeenDevice = useSelector(lastSeenDeviceSelector);
+
+  const deviceIcon = useCallback(
+    ({ color, size }: { color: string; size: number }) => {
+      let icon;
+      switch (lastSeenDevice?.modelId) {
+        case DeviceModelId.nanoS:
+        case DeviceModelId.nanoSP:
+          icon = <Icons.NanoSFoldedMedium size={size} color={color} />;
+          break;
+        case DeviceModelId.nanoFTS:
+          icon = <Icons.PowerMedium size={size} color={color} />;
+          break;
+        case DeviceModelId.nanoX:
+        default:
+          icon = <Icons.NanoXFoldedMedium size={size} color={color} />;
+          break;
+      }
+
+      return hasAvailableUpdate ? (
+        icon
+      ) : (
+        <Box>
+          {icon}
+          <Badge />
+        </Box>
+      );
+    },
+    [lastSeenDevice?.modelId, hasAvailableUpdate],
+  );
```

Comment:
Extract it to it's own component at this point?

Reply:
Good point. Fixed it.